{{cookiecutter.project_slug}}/.pre-commit-config.yaml

```diff
@@ -17,18 +17,25 @@ repos:
   - id: end-of-file-fixer
   - id: mixed-line-ending
     args: ['--fix=auto']  # replace 'auto' with 'lf' to enforce Linux/Mac line endings or 'crlf' for Windows
+  - id: detect-private-key
 
 # Ruff replaces black, flake8, autoflake, isort and more
 - repo: https://github.com/charliermarsh/ruff-pre-commit
-  rev: 'v0.5.7'  # make sure this is always consistent with hatch configs
+  rev: 'v0.8.3'  # make sure this is always consistent with hatch configs
   hooks:
     - id: ruff
     - id: ruff-format
       args: [--check, --config, ./pyproject.toml]
 
 - repo: https://github.com/pre-commit/mirrors-mypy
-  rev: 'v1.11.1'  # make sure this is always consistent with hatch configs
+  rev: 'v1.13.0'  # make sure this is always consistent with hatch configs
   hooks:
     - id: mypy
       args: ["--install-types", "--non-interactive"]
       additional_dependencies: [types-tabulate, types-cachetools]
+
+-   repo: https://github.com/jorisroovers/gitlint
```

Comment:
I really like the idea but would it be better to have this optional via cookiecutter? Gitlint is still having less than 1k Github stars and is it really used that often? I like it but it's not a commonly found best practice, is it?

Reply:
Well, to be honest: gitlint was the only commit-message linter I found. For automated generation of a changelog it is necessary to have proper commit messages. I struggled somewhat on that ... 
Probably more professional developers than I am are much more familiar with this and do it while asleep but for me it was helpful.
I do not knoy how cookiecutter would be of any help here but probably I missed something there.

Reply:
Ah, I guess now I know what you mean with "optional via cookiecutter". You mean similar to the lock-file support i. e.  controlling via y/n prompt during cookiecutter execution? I am afraid it is me lacking some cookiecutter / Python packaging experience that I did not get it directly. Sounds like a valid option.

Probably also an option fot the complete footgun-stuff ...

Reply:
Exactly, to make it gitlint an additional option. Bear in mind though that with every additional option, running unittests becomes more complicated since you need to test every possible path, not only the one that includes all features.

Reply:
Question: to do it perfect, I would have to delete the .gitlint config file after generation via the post_gen_project.py hook ?

Reply:
I tried to make the gitlint optional.
It works except two issues:
- the post-generation commit doe snot delete the .gitlint file
- the Jinja-tmeplating in pre-commit--config.yaml does not work

Reply:
Works now